### ISSUE TITLE
Wrong calculation of ballot-size truncates choices of voters

### ISSUE THREAD

**Issue by PascalinDe** [c4dt/d-voting#167]
# TLDR

A wrong encoding in the frontend cuts off the voter's input for the last candidate with elections having a multiple of 6 candidates. The voter's intention on the last candidate cannot be retrieved afterwards, and is therefore lost for good.

The erroneous source code can be found [here](https://github.com/c4dt/d-voting/blob/main/web/frontend/src/pages/ballot/components/VoteEncode.tsx#L14).

# Detailed description

To hide the voters intentions, d-voting encrypts the choice of the voters into an `Elliptic Curve Point`. This is done
using the [Embed](https://github.com/dedis/kyber/blob/master/group/edwards25519/point.go#L114) method of the dedis/kyber library, which allows to embed up to 29 bytes of data and to encrypt it later.

## Documentation 
The correct encoding of a ballot according to the [documentation](https://github.com/c4dt/d-voting/blob/main/docs/ballot_encoding.md) is confusing, as it says that the ID is in base64, while the examples are in hex.

The problem is how different parts of the code interpret this documentation.

## Dela-code

Even though the documentation says that the ID is 3 bytes, the dela-code uses 6 bytes. In the `Form` structure, which contains an election, this ID is stored base64-encoded.

When a new form is created by the frontend, the new form is sent to the Dela-code, which calculates how many `Elliptic Curve Points` are needed to encrypt a ballot, and sets this number in the form. To calculate this number, the Dela-code counts the number of bytes necessary to encode things, and uses the base64-encoded length of the 6 bytes, the `ID`, as shown here: [calculation of encoding length](https://github.com/c4dt/d-voting/blob/main/contracts/evoting/types/ballots.go#L335).

## Frontend

Unfortunately the frontend has yet another interpretation of the documentation and supposes that the `ID` present in the form is not encoded yet, so it base64-encodes the `ID`, which creates a base64 encoding of a base64-encoded bytestring.

## Summary

- the encryption is done in `Elliptic Curve Points` as multiple of 29 bytes
- the documentation and the code have different ideas of the length of the ID:
    - documentation: 3 bytes
    - dela-code: 6 bytes
 - the frontend and the dela-code interpret the `ID` field once as not encoded, once as already encoded, which leads to the following lengths of this field: 
    - `ID` length in dela-code: 8 characters
    - `ID` length added to a ballot in the frontend: 12 characters (base64 encoding of the base64 encoded 6 bytes)

# Example

When a form is created, the frontend sends the form to the dela-code, which calculates the maximum length of a ballot using the 8 characters for the ID. For 6 candidates, this gives a size of 28 bytes (there is an additional `\n`), which just fits into one `Elliptic Curve Point`:

- 0:5 - `select`
- 6 - `:`
- 7:14 - base64-encoded ID
- 15 - `:`
- 16:26 - `a,b,c,d,e,f` where each of the letters is either `0` or `1`, depending on the choice of the voter
- 27 - `\n`

However, when casting a ballot, the frontend now creates a string which has 32 bytes:
- 0:5 - `select`
- 6 - `:`
- 7:18 - base64-encoded ID
- 19 - `:`
- 20:30 - `a,b,c,d,e,f` where each of the letters is either `0` or `1`, depending on the choice of the voter
- 31 - `\n`

As the bytes 30:31 contain the `f\n` for the selection of the 6th candidate, these bytes are lost, as they are not encrypted and stored.

# Fix

- [ ] update the documentation to fit the code
- [ ] don't double-encode base64 in the frontend
- [x] correctly calculate the length of the ballot by taking into account that there is no trailing `,`, so the total length of the choices is `len(selection.Choices) * 2 - 1`
